geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/JoinProcess.scala

```diff
@@ -0,0 +1,250 @@
+  description = "Queries a feature type based on attributes from a second feature type"
+)
+class JoinProcess extends VectorProcess with LazyLogging {
+
+  /**
+    *
+    * @param primary main feature collection to query
+    * @param secondary secondary feature collection to query with results from first feature collection
+    * @param joinAttribute attribute to join on
+    * @param joinFilter additional filter to apply to joined features
+    * @param attributes attributes to return, from both collections, qualified by schema name
+    * @param bins flag to return results in BIN format
+    * @param binDtg date field for bin records - will use default date if not specified
+    * @param binTrackId track ID field for bin records - will not include trackId if not specified
+    * @param binLabel label field for bin record (optional)
+    * @param monitor listener to monitor progress
+    * @throws org.geotools.process.ProcessException if something goes wrong
+    * @return
+    */
+  @throws(classOf[ProcessException])
+  @DescribeResult(name = "result", description = "Output features")
+  def execute(@DescribeParameter(name = "primary", description = "Primary feature collection being queried", min = 1)
+              primary: SimpleFeatureCollection,
+              @DescribeParameter(name = "secondary", description = "Secondary feature collection to be joined", min = 1)
+              secondary: SimpleFeatureCollection,
+              @DescribeParameter(name = "joinAttribute", description = "Attribute field to join on", min = 1)
+              joinAttribute: String,
+              @DescribeParameter(name = "joinFilter", description = "Additional filter to apply to joined features", min = 0)
+              joinFilter: Filter,
+              @DescribeParameter(name = "attributes", description = "Attributes to return. Attribute names should be qualified with the schema name, e.g. foo.bar", min = 0, max = 128, collectionType = classOf[String])
+              attributes: java.util.List[String],
+              @DescribeParameter(name = "bins", description = "Return BIN records instead of regular records", min = 0)
+              bins: java.lang.Boolean,
+              @DescribeParameter(name = "binDtg", description = "Date field to use for BIN records", min = 0)
+              binDtg: String,
+              @DescribeParameter(name = "binTrackId", description = "Track field to use for BIN records", min = 0)
+              binTrackId: String,
+              @DescribeParameter(name = "binLabel", description = "Label field to use for BIN records", min = 0)
+              binLabel: String,
+              monitor: ProgressListener): SimpleFeatureCollection = {
+
+    import org.locationtech.geomesa.filter.ff
+
+    import scala.collection.JavaConversions._
+
+    logger.trace(s"Attempting join query on ${joinAttribute.getClass.getName}")
+
+    if (primary.isInstanceOf[ReTypingFeatureCollection] || secondary.isInstanceOf[ReTypingFeatureCollection]) {
```

Comment:
yeah historically this is for debugging other cases like problems solved by the layer name in geoserver not matching the type name in accumulo i think it would make a retyping feature collection and our visitors then wouldn't work so we looked for log statements like this :sob: